libraries/scripting/common/src/kotlin/script/experimental/api/replApi.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin.script.experimental.api
+
+import kotlin.script.experimental.util.ReplCompletionVariant
+import kotlin.script.experimental.util.ReplDiagnosticMessage
+
+typealias CompiledSnippet = CompiledScript<Any>
+typealias ReplCompletionResult = Sequence<ReplCompletionVariant>
+typealias ReplAnalyzerResult = Iterable<ReplDiagnosticMessage>
+
+interface ReplCompiler<CompiledSnippetT : CompiledSnippet> {
+    val lastCompiledSnippet: ILinkedPushStack<CompiledSnippetT>?
+
+    fun compile(
+        snippets: Iterable<SourceCode>,
+        configuration: ScriptCompilationConfiguration
+    )
+            : ResultWithDiagnostics<ILinkedPushStack<CompiledSnippetT>>
+
+    fun compile(
+        snippet: SourceCode,
+        configuration: ScriptCompilationConfiguration
+    )
+            : ResultWithDiagnostics<ILinkedPushStack<CompiledSnippetT>> = compile(listOf(snippet), configuration)
+}
+
+interface ReplCompleter {
+
+    fun complete(
+        snippet: SourceCode,
+        cursor: Int,
```

Comment:
@ligee is there a way to easily convert `SourceCode.Position` to `Int` and back? If no, I'd leave it as is, because now it seems like unnecessary complication (especially for API user)